From patchwork Wed Apr 17 04:10:26 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Nadav Amit <nadav.amit@gmail.com>
X-Patchwork-Id: 10904457
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 9169717E0
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 17 Apr 2019 04:10:34 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 7865F2848F
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 17 Apr 2019 04:10:34 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 6C50D28A3C; Wed, 17 Apr 2019 04:10:34 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-8.0 required=2.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,FREEMAIL_FROM,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI
	autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 153B428821
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 17 Apr 2019 04:10:33 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1725814AbfDQEKc (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Wed, 17 Apr 2019 00:10:32 -0400
Received: from mail-pf1-f196.google.com ([209.85.210.196]:33032 "EHLO
        mail-pf1-f196.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1725320AbfDQEKc (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 17 Apr 2019 00:10:32 -0400
Received: by mail-pf1-f196.google.com with SMTP id h5so11465527pfo.0
        for <kvm@vger.kernel.org>; Tue, 16 Apr 2019 21:10:32 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20161025;
        h=from:to:cc:subject:date:message-id:mime-version
         :content-transfer-encoding;
        bh=EpvcvJYIZ3OqE2YMwP2x1BjfCV/fHz0czkFDkDPW32M=;
        b=sRFn6D2kB05Q21abx7mYWt3hSW1brUhyD3TafLz7A3IVAK/t/IFYn0SK/u+vjDH7jl
         1W3mqSFUeDtnlGBnjD2qfj4tVMn/y1R9EgsmN+bAtOQ4XRnmSlGk01Tngf+DldYXivE4
         rYnfIwV0MqICYfR5+qdPh+/otJiRMlwsRJi+xPhH7a97XtLLwengbiNlo67eFbOPO5vJ
         LmzAEtsWMjjNHd1G7TL1JSSuhoTqOWg4moZIBQ/O4GdP6oCDRrYdgqbNFm2bGZ7r4mJx
         8jr160C0ICFPsx91qByDNNdnEY+OEpb1Sz48zJojJlAG3jjzMhOf1rs3g92a06UN8Z/V
         84nQ==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id:mime-version
         :content-transfer-encoding;
        bh=EpvcvJYIZ3OqE2YMwP2x1BjfCV/fHz0czkFDkDPW32M=;
        b=QuyNyOPE88ojbuX5ywplOd1r2+lqBBnSI1E98uh0LxH1oloBp48wMLMrzMCfRkSWiY
         oGAN9M/I6OLoxHkh8lAutw3NGacswiEPYvmWuT8/QTd+wrU5YnL6/lPfIB1qIABXd6ng
         RSUrVpQ3n1Sq0FIAoL3Dzcd6vIY4qKn6yinqeYYOPiqzi2ju9mngtwKMig9pEDWjmhhL
         QYFelabBtypMvyuyM4VDRLDUxkSz+qvPthZvVCMqXoe1SIYCHA9BVhXEWyGJKp+jSAr6
         t+LUVTwJb9/f2WxEdOEMMapDjW57k846krfF9PLSR8IES/NnjJYwd+sy4vPNG/ZBXE60
         gzXg==
X-Gm-Message-State: APjAAAX4qN3Veg7CwpCD/JjoLCACu/MV1i4ISjZmR6Zmz1ZQ2PGV3UzW
        B0jm7DKN8M69gh/U8+vW324=
X-Google-Smtp-Source: 
 APXvYqzKbi5hZ17zHnY6+zGx8NlDIaNWc6zS4MzmSRCpOPCL/DIRdC4GV2f4XXqA0Wx1+50mrbd4iw==
X-Received: by 2002:a62:424f:: with SMTP id
 p76mr87062147pfa.141.1555474230065;
        Tue, 16 Apr 2019 21:10:30 -0700 (PDT)
Received: from htb-2n-eng-dhcp405.eng.vmware.com ([66.170.99.1])
        by smtp.gmail.com with ESMTPSA id
 h189sm1331557pfc.125.2019.04.16.21.10.29
        (version=TLS1_2 cipher=ECDHE-RSA-AES128-GCM-SHA256 bits=128/128);
        Tue, 16 Apr 2019 21:10:29 -0700 (PDT)
From: nadav.amit@gmail.com
To: Paolo Bonzini <pbonzini@redhat.com>
Cc: kvm@vger.kernel.org,
        Sean Christopherson <sean.j.christopherson@intel.com>,
        Nadav Amit <nadav.amit@gmail.com>
Subject: [kvm-unit-tests PATCH] x86: Initialize vmcs_root
Date: Tue, 16 Apr 2019 21:10:26 -0700
Message-Id: <20190417041026.14274-1-nadav.amit@gmail.com>
X-Mailer: git-send-email 2.19.1
MIME-Version: 1.0
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

From: Nadav Amit <nadav.amit@gmail.com>

vmcs_root was not initialized, so vmclear was run on the memory in which
the IDT resides. That is not good.

Signed-off-by: Nadav Amit <nadav.amit@gmail.com>
---
 x86/vmx.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/x86/vmx.c b/x86/vmx.c
index f713ada..c0f4749 100644
--- a/x86/vmx.c
+++ b/x86/vmx.c
@@ -1244,6 +1244,8 @@ static void init_vmx(void)
 	vmxon_region = alloc_page();
 	memset(vmxon_region, 0, PAGE_SIZE);
 
+	vmcs_root = alloc_page();
+
 	fix_cr0_set =  rdmsr(MSR_IA32_VMX_CR0_FIXED0);
 	fix_cr0_clr =  rdmsr(MSR_IA32_VMX_CR0_FIXED1);
 	fix_cr4_set =  rdmsr(MSR_IA32_VMX_CR4_FIXED0);
